sycl/source/detail/pi.cpp

```diff
@@ -491,6 +498,13 @@ pi_uint32 DeviceBinaryProperty::asUint32() const {
   return sycl::detail::pi::asUint32(&Prop->ValSize);
 }
 
+std::vector<unsigned char> DeviceBinaryProperty::asByteArray() const {
+  assert(Prop->Type == PI_PROPERTY_TYPE_BYTE_ARRAY && "property type mismatch");
+  assert(Prop->ValSize > 0 && "property size mismatch");
+  const auto *Data = pi::cast<const unsigned char *>(Prop->ValAddr);
+  return {Data, Data + Prop->ValSize};
```

Comment:
If I'm not mistaken, this constructor will copy data. This seems unnecessary, since property lives as long as the containing image

Reply:
That's right, but I'm also not aware of any way to avoid that while using `std::vector`. Do you have any suggestions?

Reply:
don't use std::vector then :)

Reply:
I mean, if it does not allow to do what's needed here

Reply:
you may e.g. pass around the `Prop->ValAddr + sizeof(uint64_t)` pointer + size and have an API to get I'th bit instead of using vector as the middleman